### ISSUE TITLE
IPv6 not working for test-programs

### ISSUE THREAD

**Issue by uablrek** [lxin/quic#17]
Even though an ipv6 is accepted, only ipv4 is used:
```
sample_test server :: 8000 keys/server-key.pem keys/server-cert.pem
# works. But...
netstat -putln | grep 8000
udp        0      0 0.0.0.0:8000            0.0.0.0:*                           -                   
```
`AF_INET` is hard-coded in the test programs.

I also miss the program for `netstat -p`. Only a dash is printed.

**Comment by uablrek** [comment]
I build the modules out-of-tree. I will test to copy to the kernel source, as instructed in the readme. BTW I use `linux-6.12.1`.

**Comment by uablrek** [comment]
Same error when the quic module is copied to the kernel-source dir. And BTW, I *have* configured ipv6 in my kernel build.

I can see ipv6 code in [protocol.c](https://github.com/lxin/quic/blob/main/modules/net/quic/protocol.c), so it "should work" (ever heard that before?)

**Comment by uablrek** [comment]
Oh, and ipv4 works, at least `func_test` and `sample_test client addr port none none`.
I haven't figured out how to use "PSK_FILE" and "HOSTNAME" to `sample_test client`, please provide an example if possible. I generate the keys/certs:
```
> ls keys
ca-cert.pem      client-key.pem   server-cert.pem  server-req.pem
ca-key.pem       client-psk.txt   server-key.pem   server.ext
client-cert.pem  client-req.pem   server-psk.txt
```

**Comment by lxin** [comment]
Linux QUIC supports IPv6, it's just the sample_test doesn't program for ipv6.

For the ipv6 socket, it's no different from the TCP socket programming, just change to use:

socket(AF_INET6, ...);

struct sockaddr_in6 sa = {};
        
sa.sin6_family = AF_INET6;
sa.sin6_port = htons(atoi(port));
inet_pton(AF_INET6, ipv6, &sa.sin6_addr);


For PSK:

server:
./sample_test server 0.0.0.0 1234 ./keys/server-psk.txt none none
client:
./sample_test client 127.0.0.1 1234 ./keys/client-psk.txt none none


For HOSTNAME,

it's used to set Server name indication in TLS msg to tell the server which certificate should be sent:

  https://www.gnutls.org/manual/html_node/Server-name-indication.html

For example, the certificate you generated is with 'DNS.1 = server.test'.

server:
./sample_test server 0.0.0.0 1234 ./keys/server-key.pem ./keys/server-cert.pem
client:
./sample_test client 127.0.0.1 1234 server.test none


Note that there's a mistake in ./sample_test.c, and the arguments psk and host don't match the order:

./sample_test client <PEER ADDR> <PEER PORT> <PSK_FILE | 'none'> <HOSTNAME | 'none'> [ALPN]

if you want to use 'hostname', you can fix it first:

diff --git a/tests/sample_test.c b/tests/sample_test.c
index 365741a..432ba6f 100644
--- a/tests/sample_test.c
+++ b/tests/sample_test.c
@@ -37,8 +37,8 @@ static int do_client(int argc, char *argv[])
                return -1;
        }
 
-       psk  = strcmp(argv[4], "none") ? argv[4] : NULL;
-       host = strcmp(argv[5], "none") ? argv[5] : NULL;
+       host = strcmp(argv[4], "none") ? argv[4] : NULL;
+       psk  = strcmp(argv[5], "none") ? argv[5] : NULL;

Thanks.

**Comment by uablrek** [comment]
Yeah, I missed the AF_INET in the socket() calls :blush: 
Then my first observation was correct; the test-programs only support ipv4. I have started PR #18 for this.
Thanks for the reply

**Comment by uablrek** [comment]
@lxin I have everything working now for both IPv4 and IPv6. #18 is ready for review.

The parameter order in sample_test.c seems ok already. Thanks again for the pointers. I should have checked `runtest.sh` for examples, sorry about that.

**Comment by uablrek** [comment]
Fixed by #18
